cookbooks/ohai_backport/recipes/default.rb

```diff
@@ -0,0 +1,45 @@
+#
+# Cookbook Name:: ohai_backport
+# Recipe:: default
+#
+# Copyright (c) 2019-present, Facebook, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+%w{
+  /etc/chef
+  /etc/chef/ohai
+  /etc/chef/ohai/plugins
+  /etc/chef/ohai/plugins/linux
+}.each do |dir|
+  directory dir do
+    owner 'root'
+    group 'root'
+    mode '0755'
+  end
+end
+
+# for https://github.com/chef/ohai/pull/1476
+cookbook_file '/etc/chef/ohai/plugins/linux/virtualization.rb' do
```

Comment:
why bother with all of this? Just jam those 3 lines into the attributes file?

```
    if File.exist?("/proc/self/cgroup")
      cgroup_content = File.read("/proc/self/cgroup")
      # These two REs catch many different examples. Here's a specific one
      # from when it is docker and there is no subsystem name.
      # https://rubular.com/r/dV13hiU9KxmiWB
      if cgroup_content =~ %r{^\d+:[^:]*:/(lxc|docker)/.+$} ||
          cgroup_content =~ %r{^\d+:[^:]*:/[^/]+/(lxc|docker)-?.+$}
        logger.trace("Plugin Virtualization: /proc/self/cgroup indicates #{$1} container. Detecting as #{$1} guest")
        node.automatic['virtualization'][:system] = $1
        node.automatic['virtualization'][:role] = "guest"
        node.automatic['virtualization'][:systems][$1.to_sym] = "guest"
end
```

you only need this for tests to pass, who cares?!

Reply:
That almost works, but `node['virtualization']['system']` is still hyperv for some reason

Reply:
Yup I tried both `node.default` and `node.automatic` and it fails the same way

Reply:
No wait, now it's working, wtf